message/messaging_service.cc

```diff
@@ -1046,14 +1046,14 @@ future<> messaging_service::send_mutation_done(msg_addr id, unsigned shard, resp
     return send_message_oneway(this, messaging_verb::MUTATION_DONE, std::move(id), shard, std::move(response_id), std::move(backlog));
 }
 
-void messaging_service::register_mutation_failed(std::function<future<rpc::no_wait_type> (const rpc::client_info& cinfo, unsigned shard, response_id_type response_id, size_t num_failed, rpc::optional<db::view::update_backlog> backlog)>&& func) {
+void messaging_service::register_mutation_failed(std::function<future<rpc::no_wait_type> (const rpc::client_info& cinfo, unsigned shard, response_id_type response_id, size_t num_failed, rpc::optional<db::view::update_backlog> backlog, bool rate_limit_hit)>&& func) {
```

Comment:
All extensions should be added as rpc::optional<>. Also make it even more generic then enum. Something like:
```
struct error {
  struct rate_limited{}; 
  std::variant<rate_limited> reason;
}
```